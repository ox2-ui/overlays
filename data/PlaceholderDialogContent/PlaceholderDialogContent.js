import React, { PropTypes } from 'react';
import Button from '@ox2/button/Button';

const shortContent = 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Possimus quod voluptate sunt, debitis facere neque rerum sapiente. Maiores dolor saepe cupiditate officia! Quas enim recusandae corrupti perferendis alias nemo eaque explicabo minus earum veniam amet minima dolore commodi ducimus, doloremque excepturi, sapiente porro possimus iusto, ipsa, repudiandae eos? Esse, saepe.';

const longContent = 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Mollitia ut quas itaque dolor nam temporibus aspernatur, ipsum officiis omnis, suscipit dolorum numquam voluptatibus. Porro amet nam perspiciatis recusandae, veritatis atque! Quibusdam sapiente distinctio ab accusamus, qui itaque, magnam adipisci ipsam esse quasi porro voluptas necessitatibus nostrum ratione reiciendis eaque incidunt id illum iusto, possimus. Voluptas voluptatem, expedita. Voluptates explicabo nostrum corrupti aliquam at repudiandae, earum nesciunt architecto fuga. Hic fugit in laudantium facilis et tempora! Voluptatum dignissimos, beatae adipisci rem sed quam, pariatur distinctio maiores natus sapiente, ratione et esse dolor. Id hic praesentium laudantium ratione unde, facere assumenda ab cumque laborum, nostrum illo nisi, nesciunt quibusdam perferendis voluptate aspernatur ducimus porro explicabo similique distinctio delectus deserunt at. Itaque commodi, maiores rerum magni sunt quaerat dolor accusamus. Quis sed delectus et, cum culpa autem esse tenetur alias, odio voluptatem veniam eos eaque quos libero. Quia cum, perferendis sunt! Dolorum distinctio et neque reiciendis, temporibus quisquam necessitatibus consequuntur labore cupiditate? Voluptas deleniti minima nam molestias, consequuntur odit unde voluptate vero laboriosam porro! Eos, doloribus itaque tenetur excepturi sint eaque voluptate asperiores odio ipsum vitae iusto officia officiis sapiente impedit nostrum aut, ea dolores id autem reiciendis voluptates. Suscipit quibusdam consequuntur assumenda temporibus modi sit, reiciendis, dolore velit iste, iusto ex totam sapiente sunt quaerat distinctio dicta officiis similique a vero. Sed dicta esse velit saepe quis, eos labore quia facilis inventore laudantium illum, nesciunt maxime ad. Fugit vel voluptates earum, vero quae tenetur voluptatem veritatis et quo doloribus, dicta dolorum voluptate corrupti eligendi neque temporibus iure consequuntur inventore quasi repellat culpa commodi optio dolorem a! Vitae tempore pariatur omnis rerum harum nihil aliquam neque quaerat beatae obcaecati repellat illo est, odit atque, enim dignissimos, maiores magni nesciunt. At voluptas, quas dolor accusantium quisquam magnam, architecto vel. Quis consectetur quo fuga repellendus.';

const xlongContent = 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quia repudiandae amet temporibus, veniam eos quisquam. Nisi optio et eveniet molestiae explicabo quisquam, alias consequatur reiciendis placeat velit, assumenda ut, unde esse numquam dolorum voluptates eos animi incidunt ipsam. Incidunt cum id, odit nam at dicta quos error eos blanditiis! Exercitationem doloribus deleniti, hic debitis beatae alias temporibus modi impedit molestiae ab, unde minima rerum recusandae assumenda, quam sequi, nulla atque minus corporis. Magnam ducimus necessitatibus, eligendi nemo! Deleniti tempora sapiente doloribus, placeat quibusdam reprehenderit praesentium eum modi natus rerum, nobis sint, consequatur aliquam, nam eveniet quaerat optio. Doloribus eum praesentium quis. Officiis, beatae voluptas aspernatur repudiandae repellat maxime dignissimos quas rem consequuntur ea, reprehenderit nemo vitae vero repellendus quisquam eaque velit porro nihil impedit voluptatum officia molestiae facere iusto eos. Praesentium accusantium est molestias dolor nobis suscipit aliquam, necessitatibus magni, facere dolorem rerum expedita dicta mollitia fugit deserunt impedit cum, quisquam perspiciatis, modi error. Temporibus ipsam, provident quas cupiditate rerum ex, labore facilis non enim laudantium, cum dolorem tempora dignissimos veniam eius quam tempore, vitae maxime delectus. Ullam sunt suscipit earum explicabo eligendi accusamus quibusdam officia odit enim ipsam. Voluptatem nulla accusantium autem similique, doloribus, iusto excepturi sit nihil, cumque fugit iure explicabo numquam! Dolore, odio, inventore! Doloribus iusto blanditiis tempora tenetur quia animi eveniet eius sed aspernatur molestias dolorum sequi architecto deserunt adipisci vel dolor, obcaecati autem, laboriosam a maiores. Adipisci eum eligendi, iste vero odit pariatur. Exercitationem earum sequi provident facilis quidem, amet consequuntur. Ab, ducimus. Debitis mollitia dolorum dicta recusandae alias itaque blanditiis consectetur, voluptatum porro laborum, molestias, illo exercitationem aut natus ea culpa aliquam esse atque? Dolore maiores eum inventore praesentium, voluptates ipsam ratione fugiat reiciendis sunt ad. Vitae, perspiciatis aspernatur dicta, quidem, officia cupiditate similique autem possimus voluptate architecto dolorum facilis doloremque. Sequi unde, nesciunt voluptatum saepe accusamus vel, veniam, repudiandae, quae tempore distinctio dignissimos fugit. Nobis ad quod cupiditate mollitia, repellendus corporis hic, quaerat itaque aut maiores eligendi, dicta eum? Nostrum, ipsam, porro! Voluptatibus iure laudantium eos, libero illum saepe itaque consequatur commodi obcaecati facere aut doloribus ducimus expedita repudiandae, animi nihil tempore, vel rerum doloremque ut! Quo commodi molestiae corporis quas id cumque. At nesciunt neque obcaecati laboriosam accusamus dolore, odit non mollitia officiis deleniti quasi aliquam architecto voluptas tempore reprehenderit accusantium, laudantium aspernatur inventore. Voluptate atque placeat sed numquam et, quaerat quam harum culpa voluptates praesentium ad totam nemo asperiores alias nisi ex ea illum velit incidunt. Accusantium minus, numquam veniam placeat architecto repellat voluptatem alias incidunt eius natus mollitia rerum consequatur beatae id officiis expedita debitis aspernatur commodi quas neque! Et provident quod accusamus nostrum enim officiis, harum pariatur doloremque similique, culpa sequi assumenda excepturi animi nemo molestias impedit, cupiditate natus fuga voluptate facere facilis. Consequatur voluptas dolore voluptates fuga molestiae expedita pariatur eaque illo facilis omnis reprehenderit, repellendus at quod, explicabo in facere aperiam nihil eos saepe ut autem? Aliquam labore in quidem quod atque rem odio temporibus dolor veritatis sit magni, molestias repudiandae nostrum rerum voluptate dolore fugit fuga, consectetur necessitatibus illo vitae magnam quisquam ipsam sed. Accusamus ex aliquid quidem facilis, itaque rem temporibus. Voluptas magnam, pariatur reiciendis. Sit mollitia fuga voluptates ipsam enim similique, qui ducimus cum quia facere sunt dolor necessitatibus eligendi reiciendis maiores repellendus amet quisquam ipsa consequatur laboriosam minus culpa, asperiores. Porro officia adipisci sint quod quam corporis animi eveniet reiciendis maxime rem incidunt quis, harum explicabo possimus sunt hic nulla, veniam asperiores, odio architecto culpa, dolorem? Molestias veritatis dolorem minus velit impedit nostrum, facilis expedita inventore culpa nesciunt nobis perspiciatis soluta excepturi! Ab recusandae sapiente doloremque placeat ducimus sint error iste repudiandae architecto.';

/**
 * PlaceholderDialogContent Component
 */
const PlaceholderDialogContent = (props) => {
  const { className, onDismiss, size} = props;

  return (
    <div className={className}>
      <Button onClick={() => onDismiss()} btn="raised action">Done</Button>
      {do {
        if (size === 'short') {
          shortContent;
        } else if (size === 'long') {
          longContent;
        } else if (size === 'xlong') {
          xlongContent;
        }
      }}
    </div>
  );
};

PlaceholderDialogContent.propTypes = {
  /**
   * The css class name of the root element.
   */
  className: PropTypes.string,
};

export default PlaceholderDialogContent;
