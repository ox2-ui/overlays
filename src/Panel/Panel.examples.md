### Left Panel

    initialState = { isOpen: false };
    const handleOpen = () => setState({ isOpen: true });
    const handleDismiss = () => {
      window.setTimeout(() => {
        setState({isOpen: false});
      }
      , 350);
    };

    const MyContent = (props) => {
      const {onDismiss} = props;
      return (
        <div style={{background: 'white', height: '100%'}}>
          <button onClick={() => onDismiss()}>Close</button>
          <div>Content</div>
        </div>
      );
    };

    <div>
      <button onClick={() => handleOpen()}>Open</button>


      {state.isOpen &&
        <Panel
          zDepth={10}
          isOpen={state.isOpen}
          maxWidth={'320px'}
          transition={'slideFromLeft'}
          right={'auto'}
          onDismiss={handleDismiss}
        >
          <MyContent />
        </Panel>
      }

    </div>

### Fullscreen Panel

    initialState = { isOpen: false };
    const handleOpen = () => setState({ isOpen: true });
    const handleDismiss = () => {
      window.setTimeout(() => {
        setState({isOpen: false});
      }
      , 350);
    };

    const MyContent = (props) => {
      const {onDismiss} = props;
      return (
        <div style={{background: 'white', height: '100%'}}>
          <button onClick={() => onDismiss()}>Close</button>
          <div>Content</div>
        </div>
      );
    };

    <div>
      <button onClick={() => handleOpen()}>Open</button>


      {state.isOpen &&
        <Panel
          zDepth={10}
          isOpen={state.isOpen}
          hideBackdrop={true}
          hideShadow={true}
          transition={'zoomIn'}
          onDismiss={handleDismiss}
        >
          <MyContent />
        </Panel>
      }

    </div>

### Bottom Panel

    initialState = { isOpen: false };
    const handleOpen = () => setState({ isOpen: true });
    const handleDismiss = () => {
      window.setTimeout(() => {
        setState({isOpen: false});
      }
      , 350);
    };

    const MyContent = (props) => {
      const {onDismiss} = props;
      return (
        <div style={{background: 'white', height: '100%'}}>
          <button onClick={() => onDismiss()}>Close</button>
          <div>Content</div>
        </div>
      );
    };

    <div>
      <button onClick={() => handleOpen()}>Open</button>


      {state.isOpen &&
        <Panel
          zDepth={10}
          isOpen={state.isOpen}
          maxWidth={'600px'}
          centered={true}
          transition={'slideFromBottom'}
          onDismiss={handleDismiss}
        >
          <MyContent />
        </Panel>
      }

    </div>
