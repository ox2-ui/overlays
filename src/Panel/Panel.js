import React, { PropTypes, Component } from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import Backdrop from '../Backdrop';
import styled from 'styled-components';

const PanelContent = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  overflow: hidden;
  z-index: 2;
  margin: ${props => props.centered ? '0 auto' : '' };
  width: ${props => props.width ? props.width : '100%' };
  height: ${props => props.height ? props.height : '100%' };
  max-width: ${props => props.maxWidth ? props.maxWidth : '100%' };
  box-shadow: ${props => props.hideShadow ? '' : '0px 0px 3px 0px rgba(50, 50, 50, 0.7)'};
`;

/**
 * Panel Component
 */
class Panel extends Component {
  static propTypes = {
    /**
     * Bottom positioning property value in css lenght units.
     */
    bottom: PropTypes.string,
    /**
     * If set to true panel will be centered horizontaly.
     */
    centered: PropTypes.bool,
    /**
     * The content of the panel.
     */
    children: PropTypes.node,
    /**
     * The css class name of the root element.
     */
    className: PropTypes.string,
    /**
     * Height of the panel content.
     */
    height: PropTypes.string,
    /**
     * Controls whether the panel has hideBackdrop or not.
     */
    hideBackdrop: PropTypes.bool,
    /**
     * Disables the default shadow around panel.
     */
    hideShadow: PropTypes.bool,
    /**
     * Controls whether the panel can be closed by clicking on overlay.
     */
    isModal: PropTypes.bool,
    /**
     * Controls whether the panel is opened or not.
     */
    isOpen: PropTypes.bool.isRequired,
    /**
     * Left positioning property value in css lenght units
     */
    left: PropTypes.string,
    /**
     * Max width of the content
     */
    maxWidth: PropTypes.string,
    /**
     * Callback function fired when the Dialog is dismissed.
     */
    onDismiss: PropTypes.func,
    /**
     * Callback function fired when the Dialog is dismissed and action submitted.
     */
    onSubmit: PropTypes.func,
    /**
     * Right positioning property value in css lenght units
     */
    right: PropTypes.string,
    /**
     * Top positioning property value in css lenght units
     */
    top: PropTypes.string,
    /**
     * Transition style
     */
    transition: PropTypes.oneOf(['zoomIn', 'slideFromTop', 'slideFromBottom', 'fadeIn', 'none', 'slideFromLeft', 'slideFromRight']).isRequired,
    /**
     * Width of the content
     */
    width: PropTypes.string,
    /**
     * This number represents the zDepth of the panel.
     */
    zDepth: PropTypes.number,
  };

  state = {
    isOpen: this.props.isOpen || false,
  };

  componentWillReceiveProps= (nextProps) => {
    if (nextProps.isOpen !== this.state.isOpen) {
      this.setState({ isOpen: nextProps.isOpen });
    }
  }

  handleSubmit = (value) => {
    if (this.props.onSubmit) {
      this.props.onSubmit({ value });
    }
    this.setState({isOpen: false});
  };

  handleDissmis = () => {
    if (this.props.onDismiss) {
      this.props.onDismiss();
    }
    this.setState({isOpen: false});
  };

  render() {
    const { className, children, centered, hideShadow, zDepth, hideBackdrop, transition, isModal, maxWidth, width, height, top, left, right, bottom } = this.props;

    return (
      <div className="pos:absolute-0" style={{zIndex: zDepth}}>
        <ReactCSSTransitionGroup
          transitionName={`transition:dialog-${transition}`}
          transitionAppear={true}
          transitionEnter={false}
          transitionAppearTimeout={400}
          transitionLeaveTimeout={350}
        >
          {this.state.isOpen &&
            <PanelContent
              key={1}
              className={className}
              hideShadow={hideShadow}
              maxWidth={maxWidth}
              width={width}
              height={height}
              centered={centered}
              style={{top: top, left: left, bottom: bottom, right: right}}
            >
              { React.cloneElement(children, { onDismiss: this.handleDissmis, onSubmit: this.handleSubmit }) }
            </PanelContent>
          }
        </ReactCSSTransitionGroup>
        { !hideBackdrop &&
          <Backdrop
            isOpen={this.state.isOpen}
            onClick={isModal ? null : this.handleDissmis}
          />
        }
      </div>
    );
  }
}


export default Panel;


