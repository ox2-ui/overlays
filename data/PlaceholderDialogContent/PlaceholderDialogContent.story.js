import React from 'react';
import { storiesOf } from '@kadira/storybook';
import PlaceholderDialogContent from './PlaceholderDialogContent';

storiesOf('PlaceholderDialogContent', module)
  .add('default', () => (
    <PlaceholderDialogContent />
  ));
