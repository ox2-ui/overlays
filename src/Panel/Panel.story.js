import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Panel from './Panel';
import Scroller from '@ox2/scroller/Scroller';
import PlaceholderDialogContent from '../../data/PlaceholderDialogContent';

storiesOf('Panel', module)
  .add('left panel', () => (
    <Panel
      zDepth={10}
      isOpen={true}
      maxWidth={'320px'}
      width={'93%'}
      transition={'slideFromLeft'}
      right={'auto'}
      className={'color:white'}
    >
      <Scroller className="padding-b:40">
        <PlaceholderDialogContent size="long" />
      </Scroller>
    </Panel>
  ))
  .add('left panel2 ', () => (
    <div>
      <button className="pos:absolute-0" style={{zIndex: 9}}>some</button>
      <Panel
        zDepth={10}
        isOpen={true}
        maxWidth={'320px'}
        width={'93%'}
        transition={'slideFromLeft'}
        right={'auto'}
        className={'color:white'}
      >
        <Scroller className="padding-b:40">
          <PlaceholderDialogContent size="long" />
        </Scroller>
      </Panel>
    </div>
  ))
  .add('right panel', () => (
    <Panel
      zDepth={10}
      isOpen={true}
      maxWidth={'320px'}
      width={'93%'}
      transition={'slideFromRight'}
      left={'auto'}
      className={'color:white'}
    >
      <Scroller className="padding-b:40">
        <PlaceholderDialogContent size="long" />
      </Scroller>
    </Panel>
  ))
  .add('right panel • modal', () => (
    <Panel
      zDepth={10}
      isOpen={true}
      maxWidth={'320px'}
      width={'93%'}
      isModal={true}
      transition={'slideFromRight'}
      left={'auto'}
      className={'color:white'}
    >
      <Scroller className="padding-b:40">
        <PlaceholderDialogContent size="long" />
      </Scroller>
    </Panel>
  ))
  .add('fullscreen', () => (
    <Panel
      zDepth={10}
      isOpen={true}
      hideBackdrop={true}
      transition={'zoomIn'}
      className={'color:white'}
    >
      <Scroller className="padding-b:40">
        <PlaceholderDialogContent size="long" />
      </Scroller>
    </Panel>
  ))
  .add('fullscreen • slideFromBottom', () => (
    <Panel
      zDepth={10}
      isOpen={true}
      hideBackdrop={true}
      transition={'slideFromBottom'}
      className={'color:white'}
    >
      <Scroller className="padding-b:40">
        <PlaceholderDialogContent size="long" />
      </Scroller>
    </Panel>
  ))
  .add('maxWidth centered • slideFromBottom', () => (
    <Panel
      zDepth={10}
      isOpen={true}
      maxWidth={'600px'}
      centered={true}
      transition={'slideFromBottom'}
      className={'color:white'}
    >
      <Scroller className="padding-b:40">
        <PlaceholderDialogContent size="long" />
      </Scroller>
    </Panel>
  ))
  .add('bottom left', () => (
    <Panel
      zDepth={10}
      isOpen={true}
      maxWidth={'320px'}
      width={'93%'}
      height={'500px'}
      transition={'slideFromRight'}
      left={'auto'}
      top={'auto'}
      className={'color:white'}
    >
      <Scroller className="padding-b:40">
        <PlaceholderDialogContent size="long" />
      </Scroller>
    </Panel>
  ))
  .add('right panel • top spacing', () => (
    <Panel
      zDepth={10}
      isOpen={true}
      maxWidth={'320px'}
      width={'93%'}
      transition={'slideFromRight'}
      left={'auto'}
      top={'20px'}
      className={'color:white'}
    >
      <Scroller className="padding-b:40">
        <PlaceholderDialogContent size="long" />
      </Scroller>
    </Panel>
  ));
