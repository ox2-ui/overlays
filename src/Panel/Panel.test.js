/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Panel from './Panel';
import Scroller from '@ox2/scroller/Scroller';
import PlaceholderDialogContent from '../../data/PlaceholderDialogContent';

it('renders correctly', () => {
  const wrapper = mount(
    <Panel
      zDepth={10}
      isOpen={true}
      maxWidth={'320px'}
      width={'93%'}
      transition={'slideFromLeft'}
      right={'auto'}
      className={'color:white'}
    >
      <Scroller className="padding-b:40">
        <PlaceholderDialogContent size="long" />
      </Scroller>
    </Panel>
  );

  expect(wrapper).toMatchSnapshot();
});
