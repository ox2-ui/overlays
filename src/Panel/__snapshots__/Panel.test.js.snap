exports[`test renders correctly 1`] = `
<Panel
  className="color:white"
  isOpen={true}
  maxWidth="320px"
  right="auto"
  transition="slideFromLeft"
  width="93%"
  zDepth={10}>
  <div
    className="pos:absolute-0"
    style={
      Object {
        "zIndex": 10,
      }
    }>
    <ReactCSSTransitionGroup
      transitionAppear={true}
      transitionAppearTimeout={400}
      transitionEnter={false}
      transitionLeave={true}
      transitionLeaveTimeout={350}
      transitionName="transition:dialog-slideFromLeft">
      <ReactTransitionGroup
        childFactory={[Function]}
        component="span"
        transitionAppear={true}
        transitionAppearTimeout={400}
        transitionEnter={false}
        transitionLeave={true}
        transitionLeaveTimeout={350}
        transitionName="transition:dialog-slideFromLeft">
        <span>
          <ReactCSSTransitionGroupChild
            appear={true}
            appearTimeout={400}
            enter={false}
            leave={true}
            leaveTimeout={350}
            name="transition:dialog-slideFromLeft">
            <styled.div
              className="color:white"
              maxWidth="320px"
              style={
                Object {
                  "bottom": undefined,
                  "left": undefined,
                  "right": "auto",
                  "top": undefined,
                }
              }
              width="93%">
              <div
                className="color:white gZvZQA"
                style={
                  Object {
                    "bottom": undefined,
                    "left": undefined,
                    "right": "auto",
                    "top": undefined,
                  }
                }
                width="93%">
                <Scroller
                  className="padding-b:40"
                  onDismiss={[Function]}
                  onSubmit={[Function]}>
                  <div
                    className="padding-b:40"
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "WebkitOverflowScrolling": "touch",
                        "backgroundColor": "transparent",
                        "bottom": "0",
                        "left": "0",
                        "overflowX": "hidden",
                        "overflowY": "auto",
                        "paddingBottom": "40px",
                        "position": "absolute",
                        "right": "0",
                        "top": "0",
                      }
                    }>
                    <PlaceholderDialogContent
                      onDismiss={[Function]}
                      onSubmit={[Function]}
                      size="long">
                      <div>
                        <Button
                          btn="raised action"
                          onClick={[Function]}>
                          <button
                            data-btn="raised action"
                            onClick={[Function]}>
                            Done
                          </button>
                        </Button>
                        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Mollitia ut quas itaque dolor nam temporibus aspernatur, ipsum officiis omnis, suscipit dolorum numquam voluptatibus. Porro amet nam perspiciatis recusandae, veritatis atque! Quibusdam sapiente distinctio ab accusamus, qui itaque, magnam adipisci ipsam esse quasi porro voluptas necessitatibus nostrum ratione reiciendis eaque incidunt id illum iusto, possimus. Voluptas voluptatem, expedita. Voluptates explicabo nostrum corrupti aliquam at repudiandae, earum nesciunt architecto fuga. Hic fugit in laudantium facilis et tempora! Voluptatum dignissimos, beatae adipisci rem sed quam, pariatur distinctio maiores natus sapiente, ratione et esse dolor. Id hic praesentium laudantium ratione unde, facere assumenda ab cumque laborum, nostrum illo nisi, nesciunt quibusdam perferendis voluptate aspernatur ducimus porro explicabo similique distinctio delectus deserunt at. Itaque commodi, maiores rerum magni sunt quaerat dolor accusamus. Quis sed delectus et, cum culpa autem esse tenetur alias, odio voluptatem veniam eos eaque quos libero. Quia cum, perferendis sunt! Dolorum distinctio et neque reiciendis, temporibus quisquam necessitatibus consequuntur labore cupiditate? Voluptas deleniti minima nam molestias, consequuntur odit unde voluptate vero laboriosam porro! Eos, doloribus itaque tenetur excepturi sint eaque voluptate asperiores odio ipsum vitae iusto officia officiis sapiente impedit nostrum aut, ea dolores id autem reiciendis voluptates. Suscipit quibusdam consequuntur assumenda temporibus modi sit, reiciendis, dolore velit iste, iusto ex totam sapiente sunt quaerat distinctio dicta officiis similique a vero. Sed dicta esse velit saepe quis, eos labore quia facilis inventore laudantium illum, nesciunt maxime ad. Fugit vel voluptates earum, vero quae tenetur voluptatem veritatis et quo doloribus, dicta dolorum voluptate corrupti eligendi neque temporibus iure consequuntur inventore quasi repellat culpa commodi optio dolorem a! Vitae tempore pariatur omnis rerum harum nihil aliquam neque quaerat beatae obcaecati repellat illo est, odit atque, enim dignissimos, maiores magni nesciunt. At voluptas, quas dolor accusantium quisquam magnam, architecto vel. Quis consectetur quo fuga repellendus.
                      </div>
                    </PlaceholderDialogContent>
                  </div>
                </Scroller>
              </div>
            </styled.div>
          </ReactCSSTransitionGroupChild>
        </span>
      </ReactTransitionGroup>
    </ReactCSSTransitionGroup>
    <Backdrop
      isOpen={true}
      onClick={[Function]}>
      <ReactCSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={700}
        transitionEnter={false}
        transitionLeave={true}
        transitionLeaveTimeout={400}
        transitionName="transition:overlay-fadeIn">
        <ReactTransitionGroup
          childFactory={[Function]}
          component="span"
          transitionAppear={true}
          transitionAppearTimeout={700}
          transitionEnter={false}
          transitionLeave={true}
          transitionLeaveTimeout={400}
          transitionName="transition:overlay-fadeIn">
          <span>
            <ReactCSSTransitionGroupChild
              appear={true}
              appearTimeout={700}
              enter={false}
              leave={true}
              leaveTimeout={400}
              name="transition:overlay-fadeIn">
              <styled.div
                onClick={[Function]}
                onTouchMove={[Function]}>
                <div
                  className="kQeSPN"
                  onClick={[Function]}
                  onTouchMove={[Function]} />
              </styled.div>
            </ReactCSSTransitionGroupChild>
          </span>
        </ReactTransitionGroup>
      </ReactCSSTransitionGroup>
    </Backdrop>
  </div>
</Panel>
`;
